sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/consumer_client.hpp

```diff
@@ -29,9 +32,17 @@ namespace Azure { namespace Messaging { namespace EventHubs {
      */
     Azure::Core::Http::Policies::RetryOptions RetryOptions{};
 
-    /**@brief  Message sender options.
-     */
-    Azure::Core::Amqp::_internal::MessageReceiverOptions ReceiverOptions{};
+    /** @brief Maximum message size for messages being sent. */
+    Azure::Nullable<std::uint64_t> MaxMessageSize;
```

Comment:
Just curious, why does this have to be nullable (not pushing back on it being so)?

Reply:
Because we want to distinguish between "the customer set this value" and "the customer did not set this value". `Azure::Nullable` clearly expresses that concept.

Theoretically (although it would be silly), 0 is a valid value for MaxMessageSize, so we can't use a sentinal value like 0 or -1.